File: apm-init/python/Agent.py
from apm.APMAgent import APMAgent


class Agent:
    def __init__(self):
        self.apmAgent = APMAgent()
        pass

    def run(self, input, saveconfig):
        print("input", input)
        output = {
            "text": "Hello APM!"
        }

        self.apmAgent.save_output(saveconfig=saveconfig, output=output) # save output

        return output

File: apm-init/nodejs/src/Agent.js
import { APMAgent } from '@fynal-ai/apm';

class Agent {
	constructor() {
		this.apmAgent = new APMAgent();
	}
	async run(input, saveconfig) {
		console.log('Receive', input);

		const text = { text: input.prompt };

		// send to ChatGPT
		const responseJSON = {};

		const output = { text: Math.random().toString().substring(2, 15) };

		this.apmAgent.saveOutput(saveconfig, output);

		return output;
	}
}

export { Agent };
